x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -84,24 +93,33 @@ static void getMlNodeStats(String[] mlNodes, Client client, TimeValue timeout, A
     }
 
     static void getMemoryAndCpu(
-        ClusterState clusterState,
+        MlAutoscalingContext autoscalingContext,
         MlMemoryTracker mlMemoryTracker,
         Map<String, OsStats> osStatsPerNode,
         ActionListener<MlAutoscalingStats> listener
     ) {
         Set<String> nodesWithRunningJobs = new HashSet<>();
-        long memoryBytesSum = osStatsPerNode.values()
+
+        // If the ML nodes in the cluster have different sizes, return 0.
```

Comment:
There are basically just 3 sizes: 4, 8, 32. The scaler by desing only allows the _same_ size for _all_ nodes.

So how can it than happen that they are different?

This can only happen when switching the type for the period until the cluster settled again. So imagine we have 1 4GB node, next we deploy a large model that doesn't fit on a 4GB node, but requires 8GB. If that's the case all 4 GB nodes get replaced by a number of 8GB nodes. Not the same number ideally, but as much as required. The nodes are not shutdown straight away, but smoothly go up(8GB) and down(4GB). The cluster transitions to a new state. During that phase this metric collector is supposed to return `0` for this value in question to signal the scaler the transition. Eventually the transition finishes. If all nodes are 8GB it will return the proper value. 

That's the design we came up to begin with. I will give you more details tomorrow(pm).

Reply:
Ah ok thanks!